docs/source/examples/checkpointer.rst

```diff
@@ -0,0 +1,319 @@
+.. _understand_checkpointer:
+
+==============================
+Understanding the Checkpointer
+==============================
+
+This tutorial will walk you through the design and behavior of the checkpointer and associated
+utilities.
+
+
+Overview
+--------
+
+TorchTune checkpointers are designed to be composable components which can be plugged
+into any recipe - training, evaluation or generation. Each checkpointer supports a
+set of models and scenarios making these easy to understand, debug and extend.
+
+TorchTune is designed to be "state-dict invariant".
+
+- At the input, TorchTune accepts checkpoints from multiple sources in multiple formats.
```

Comment:
So one possible suggestion: do we wanna explicitly state the problem that this is solving up front? Like give an example of the usual `.load_state_dict()` flow and how it doesn't work when the model and weights are coming from different places, then show how the checkpointer allows us to just call `load_checkpoint()` without having to worry about it. (Btw I think the way you've laid it out is good too, this'd just be a slightly different exposition.)

Reply:
Great idea! Let me try to put an example of this